### PR TITLE
TC #2 Contact Recommendation: Data Processing Pipeline

### PR DESCRIPTION
**Pull request by yawbtng**
## Description

**This PR...**
- Implements batch data fetching from Supabase for users and connections with proper join table handling
- Creates embedding service with LangChainJS + Together AI integration for profile text composition
- Builds batch processing pipeline for efficient vector generation and Qdrant storage
- Adds robust validation and error handling for embedding generation and vector point creation
- Implements metadata preparation for both user and connection profiles with many-to-many relationship support

**Later PRs...**
- Incremental updates via Supabase edge functions for real-time vector synchronization
- Frontend integration for recommendation display and user interaction
- Performance optimization and caching layer for recommendation queries

**Rationale**
The data processing pipeline bridges the gap between our relational database (Supabase) and vector database (Qdrant) by efficiently fetching profile data, generating semantic embeddings using Together AI, and storing them with proper metadata for similarity search. The batch processing approach ensures scalability while the validation layer maintains data quality.

## Milestones
This works towards implementing the Contact Recommendation feature by establishing the complete data pipeline needed to populate the vector database with user and connection embeddings for similarity-based recommendations.

## Resources
- https://js.langchain.com/docs/integrations/text_embedding/togetherai/
- https://qdrant.tech/documentation/tutorials/neural-search/
- https://js.langchain.com/docs/introduction/
- https://docs.together.ai/reference/embeddings
- https://supabase.com/docs/reference/javascript/select

## Test Plan
- Code will be tested via browser console and dedicated test pages in the next PR
- Batch processing handles edge cases like empty data, failed embeddings, and invalid vector points
- Validation ensures proper metadata structure and vector dimensions before Qdrant storage

### CODE REVIEW

File: Lynk/backend/services/batch-embedding-pipeline.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+import { upsertVectorsBatch } from "./qdrant-client.js";
+
+export async function processAllUserEmbeddings() {
+  const users = await fetchAllUsers();
+  if (!users.length) {
+    console.log("No users found.");
+    return;
+  }
+
+  const batchResult = await generateUserEmbeddingsBatch(users);
```
Comment (zhemeta) [inline Lynk/backend/services/batch-embedding-pipeline.js:12]: Not for this diff, just feel this could be a good use case for `Promise.all` https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all

Basically use can use `Promise` for async operations. So you don't need to set `success/failure` for each user item. 
Promise all can use to handle a list of users promises, and you can use it to get the final request result.

File: Lynk/backend/services/batch-embedding-pipeline.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  }
+
+  const vectorPoints = [];
+  batchResult.results.forEach((result, idx) => {
+    if (result.success) {
+      try {
+        const metadata = prepareUserMetadata(users[idx]);
+        const point = createVectorPoint(result.embedding, metadata);
+        const valid = validateVectorPoint(point);
+        if (valid.success) {
```
Comment (zhemeta) [inline Lynk/backend/services/batch-embedding-pipeline.js:25]: With promise.all, you don't need to check each valid success/failure in the forEach. It could be something like:
```
Promise.all([valid1, valid2, valid3])  // each valid is a promise
  .then(results => {
    // all promises fulfilled
    console.log(results); // [result_of_valid1, result_of_valid2, result_of_valid3]
  })
  .catch(error => {
    // at least one promise rejected
    console.error(error);
  });
```
